platform/ui/src/components/SidePanel/SidePanel.tsx

```diff
@@ -112,40 +103,29 @@ const SidePanel = ({
     }
   }, [swiper]);
 
-  const updatePanelOpen = useCallback((panelOpen: boolean) => {
-    setPanelOpen(panelOpen);
-    if (panelOpen) {
-      setHasBeenOpened(true);
+  useEffect(() => {
+    if (panelOpen && onSidePanelOpen) {
+      onSidePanelOpen();
     }
-  }, []);
+  }, [panelOpen, onSidePanelOpen]);
```

Comment:
can't you run your callback on the onClick?

```
onClick={() => {

              setPanelOpen(prev => !prev);
              // if it was closed, mean it will be open so
              if (!prev){
                onSidePanelOpen
              }
              
            }}
            ```

Reply:
`prev` would be `undefined` in this example and previous `updatePanelOpen` was expecting a boolean but `updatePanelOpen(prev => !prev)` was passing a function and any `if(prev)` was `true` and `onSidePanelOpen` was being called even when closing the panel.

Reply:
I've added `updatePanelOpen` back, updated both `onClick` event handlers and `onSidePanelOpen` is being called immediately with no extra `useEffect`